vnext/Microsoft.ReactNative.Cxx/CoreApp.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+#include <roapi.h>
+
+#ifdef __cplusplus
+#define NORETURN [[noreturn]]
+
+struct RNCoreApp {
```

Comment:
>RNCoreApp [](http://example.com/codeflow?start=7&length=9)

I wonder where the term "CoreApp" came from? It sounds too generic and does not explain the purpose of it.